doc/developer-notes.md

```diff
@@ -408,6 +408,34 @@ GUI
     should not interact with the user. That's where View classes come in. The converse also
     holds: try to not directly access core data structures from Views.
 
+Subtrees
+----------
+
+Several parts of the repository are subtrees of software maintained elsewhere.
+
+ Some of these are maintained by active developers of Bitcoin Core, in which case changes should probably go
+directly upstream without being PRed directly against the project.  They will be merged back in the next
+subtree merge.
+
+ Others are external projects without a tight relationship with our project.  Changes to these should also
+be sent upstream but bugfixes may also be prudent to PR against Bitcoin Core so that they can be integrated
+quickly.  Cosmetic changes should be purely taken upstream.
+
+Current subtrees include:
```

Comment:
Maybe link to https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#git-subtree-checksh, so we don't have to keep track to update 2 places whenever something changes?